docroot/modules/humsci/hs_layouts/hs_layouts.module

```diff
@@ -113,3 +114,41 @@ function hs_layouts_preprocess_pattern_vertical_link_card(&$variables) {
     $variables['title_has_heading'] = _hs_layouts_field_has_title($field);
   }
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function hs_layouts_form_entity_view_display_layout_builder_form_alter(&$form) {
+  $form['#validate'][] = '_hs_layouts_unique_main_content_section_validate';
+}
+
+/**
+ * Exactly one section should be set as 'Main Content' in Layout Builder form.
+ */
+function _hs_layouts_unique_main_content_section_validate(&$form, FormStateInterface $form_state) {
```

Comment:
Is there any way this be done in the validation plugin? It's basically duplicated by the plugin

Reply:
`entity_view_display` view display entities don't have contraints implemented yet. It's [planned for Drupal 11](https://www.drupal.org/project/drupal/issues/2869792) but not available yet. For now, the only possible way we found for validating this was to do it on the form.